lib/rbac/filterer.rb

```diff
@@ -221,7 +221,9 @@ def search(options = {})
       end
 
       if search_filter && targets && (!exp_attrs || !exp_attrs[:supported_by_sql])
-        rejects     = targets.reject { |obj| matches_search_filters?(obj, search_filter, tz) }
+        rejects = TimeProfile.with_cached_time_profile_ids do
```

Comment:
It seems like this will always query all the time profiles whether we need them or not. Not sure if that's intentional or not or whether the cost is negligible.